packages/web3torrent/src/library/paid-streaming-extension.ts

```diff
@@ -91,19 +94,31 @@ export abstract class PaidStreamingExtension implements Extension {
   stop() {
     if (!this.isForceChoking) {
       this.isForceChoking = true;
-      this.executeExtensionCommand(PaidStreamingExtensionNotices.STOP, this.pseChannelId);
+      setTimeout(() => {
+        this.executeExtensionCommand(PaidStreamingExtensionNotices.STOP, this.pseChannelId);
+      }, 0);
```

Comment:
Any clue why we need this?

Reply:
This was actually added on the previous PR about Payments, and I still quite don't get why it's necessary. It's about clearing the call stack (to respond to the requests in time) before letting the leecher know that it should stop. But I don't know if it's still necessary.

Reply:
What happen if we undo this `setTimeout()`?

Reply:
if we run the code one tab beside the other, the download never finishes, it gets stuck at the beginning. I've got yet to test what happens if run in multiple browsers, but that first one seemed like a good enough reason to leave it as is